icu4c/source/i18n/formattedval_sbimpl.cpp

```diff
@@ -96,6 +96,11 @@ void FormattedValueStringBuilderImpl::getAllFieldPositions(FieldPositionIterator
     }
 }
 
+void FormattedValueStringBuilderImpl::reset() {
+    fString.clear();
+    spanIndicesCount = 0;
```

Comment:
does it make sense to call it "reset"? If it's functionality is truely to "reset" FormattedValueStringBuilderImpl then should it also clear out the  fNumericField and spanIndices too? If fNumericField and spanIndices should not be clear out, then maybe the name "reset" does not truly reflect what you like it to do.

Reply:
Also I see there are subclass adding more fields to the object, so if it is named "reset" I would expect the function also reset the extra fields the subclass added. Maybe call it resetXXX to "limit" what the scope it intend to reset of it's member data?